src/main/java/com/myhome/services/springdatajpa/UserSDJpaService.java

```diff
@@ -68,24 +67,22 @@
 
   @Override
   public Optional<UserDto> getUserDetails(String userId) {
-    User user = userRepository.findByUserId(userId);
-    if(user != null) {
+    Optional<User> user = userRepository.findByUserId(userId);
+    return user.map(admin -> {
       Set<String> communityIds = communityService.listAll().stream()
-          .filter(community -> community.getAdmins()
-              .stream()
-              .map(CommunityAdmin::getAdminId)
-              .collect(Collectors.toSet())
-              .contains(userId)
-          )
-          .map(Community::getCommunityId)
-          .collect(Collectors.toSet());
+      .filter(community -> community.getAdmins()
+      .stream()
```

Comment:
formatting

Reply:
I don't understand what needs to be changed here

Reply:
There is a nested stream into first one.
It should be tabbed out like in previous implementation (right now all lines starts with same line starting point).
Let's move also `.stream()` from line 74 to 73